packages/core-data/src/actions.js

```diff
@@ -166,35 +166,37 @@ export function receiveEmbedPreview( url, preview ) {
  * @param {Object}   [options]                 Delete options.
  * @param {Function} [options.__unstableFetch] Internal use only. Function to
  *                                             call instead of `apiFetch()`.
- *                                             Must return a control descriptor.
+ *                                             Must return a promise.
  */
-export function* deleteEntityRecord(
+export const deleteEntityRecord = (
 	kind,
 	name,
 	recordId,
 	query,
-	{ __unstableFetch = null } = {}
-) {
-	const entities = yield getKindEntities( kind );
+	{ __unstableFetch = triggerFetch } = {}
```

Comment:
Just a note so that we don't forget.

Right now `__unstableFetch` is so named because it's a "private" argument used only by the batch action. It would be nicer, I think, to remove the argument in favour of currying `deleteEntityRecord`, `saveEntityRecord`, etc. and only exporting a partial application.

We can't easily do this until everything uses thunks.

```js
const createDeleteEntityRecordAction = ( fetch ) => ( kind, name, recordId, query ) => {
	...
};
export const deleteEntityRecord = createDeleteEntityRecordAction( triggerFetch );
```

Reply:
I really like that idea, it has some real Haskell vibes to it! :-)

Reply:
I paid $20k for a computer science degree and now I know how to use words like "currying" in a sentence.